fairlearn/datasets/_synthetic.py

```diff
@@ -0,0 +1,82 @@
+# Copyright (c) Fairlearn contributors.
+# Licensed under the MIT License.
+import numpy as np
+from sklearn.datasets import make_classification
+from sklearn.utils import check_random_state
+
+
+def make_synthetic_dataset(feature_config=None, n_features=20, n_informative=4, random_state=None):
+    """Create a synthetic dataset with a single sensitive feature: 'gender'.
+
+    Parameters
+    ----------
+    feature_config : dict, default=None
+        A dict whose keys are the labels for the gender feature, and whose set
+        the :code:`n_samples` and :code:`class_sep` parameters for :code:`make_classification`.
+        If `None`, the following dictionary is used by default::
+
+            {
+                'Man': {'n_samples': 500, 'class_sep': 1},
+                'Other': {'n_samples': 500, 'class_sep': 0.5},
+                'Unspecified': {'n_samples': 500, 'class_sep': 0.5},
+                'Woman': {'n_samples': 500, 'class_sep': 2},
+            }
+
+    n_features : int, default=20
+        The total number of features.
+
+    n_informative : int, default=4
+        The number of informative features.
+
+    random_state : int or RandomState instance, default=None
+        The random number generator seed or :class:`numpy.random.RandomState` to use.
+
+    Returns
+    -------
+    (X, y, gender) : tuple of numpy.ndarray
+        X : numpy.ndarray
+            The generated samples.
+        y : numpy.ndarray
+            Labels for the binary classification.
+        gender : numpy.ndarray
+            The sensitive feature label.
+    """
+    rng = check_random_state(random_state)
+
+    classification_kwargs = {
+        'n_features': n_features,
+        'n_informative': n_informative,
+        'n_classes': 2,
```

Comment:
I feel like we should specify somewhere (either in function name or docstring) that this function produces a binary classification dataset.  Are there any plans to incorporate multi-class synthetic datasets in the future?

Reply:
I suspect we'll have a separate function for multiclass, similar to what scikit-learn does. I'm all in favor of clarity, of course, so including "binary" in the name sounds very reasonable to me.

Reply:
I've renamed the function to `make_sensitive_classification` and expanded the docstring.